improve pair selector component

dcts opened the pull request:
Fixes #223.

Improved the pair selector by:
- replacing slash `/` and magnifying glass with chevron (as designed by @Kafkafrate)
- removed margin and made hover state take up all width (as mentioned in #223)
- sorted the coins by following algorithm: XRD/xUSDC, DEXTR/XRD, (rest sorted alphabetically)
- added coin icon to dropdown and emphasized the first token, making the second token less prominent (inspired by Binance)
- added hover state (70% opacity -> 100%opcity on hover)

# Before
![Bildschirmfoto vom 2024-02-20 00-26-56](https://github.com/DeXter-on-Radix/website/assets/44790691/e60cf076-e388-4fa5-aa7d-6ae242590c6c)

# After
![Bildschirmfoto vom 2024-02-20 00-26-37](https://github.com/DeXter-on-Radix/website/assets/44790691/5f7454c2-fa9e-493f-bae0-1ef14f61a7a3)

EvgeniiaVak reviewed (changes requested):
@dcts maybe we can replace the `+` sign at the right with token icons? (both of them) I'm not sure if that was @Kafkafrate 's intention but maybe it was just a placeholder for the icons?

Without the backslash people won't know there is this shortcut for search (or did you remove the shortcut? I vote to get it back, I think it's a useful feature. Or replace with some other shortcut. Maybe we need to adjust its design, but I think it needs to be there.

dcts commented:
We discussed in Telegram to remove the `+` for now (updated the PR accordingly). We also discussed to replace it with a trading related metric, e.g. "liquidity of the trading pair". That would provide additional information for the user. But since the design and final decision is pending, I would track that in another issue.

Concerning the `/`: I personally think its rather confusing than helping. Until today I did not even realized that there is a possibility to search coins by pressing `/`. But the functionality is pretty cool tbh, I like it a lot (and did not remove it, so its still functioning). But I'd much rather have it as "easter egg" or hidden functionality, as its not vital for the overall user experience.

But thats just my opinion, happy to change if overall community sentiment is supportive of keeping it :)

nguvictor commented:
![image](https://github.com/DeXter-on-Radix/website/assets/3978285/d21fb6ed-f9d4-4bf0-ba4c-7e265f99677f)

I think we agreed on using the Dexter pair as the first pair as it had the most volume on the chart.
![image](https://github.com/DeXter-on-Radix/website/assets/3978285/711ed76e-d391-498e-8adc-b57fd3524a2d)
VS
![image](https://github.com/DeXter-on-Radix/website/assets/3978285/e68fb8fe-23d5-4b61-b61e-3fa45716a438)

Kafkafrate commented:
I updated the dropdown in figma to include the search option for token. Should look better now. 

<img width="484" alt="Screenshot 2024-02-20 at 16 09 38" src="https://github.com/DeXter-on-Radix/website/assets/138138287/8c616e62-1a8a-4177-b3af-9c2118c71dab">

dcts commented:
@EvgeniiaVak those examples you sent make a lot of sense for those sites, as they both have a main search bar at the top of the site. On dexter we don't have that, the input only appears once we click the pair selector, so its rather hidden.

I am not a fan of adding a tooltip on hover, as at that point, the cursor is already hovering over the pair, and with a single click the search input opens. Its faster than the shortcut. It's still usefull for educating the user, but the tooltip will always appear there and its a bit unneccessary.

A new suggestion might be to add a permanent hint in the main bar at the top next to the `CONNECT` button, for example like this (a designer would need to properly mock this up):
<img width="534" alt="Bildschirmfoto 2024-02-20 um 19 31 30" src="https://github.com/DeXter-on-Radix/website/assets/44790691/9b4aac63-fd20-442e-b004-4caf57ac3f65">

This hint could also be positioned in the center, although might be too prominent.
<img width="765" alt="Bildschirmfoto 2024-02-20 um 19 22 02" src="https://github.com/DeXter-on-Radix/website/assets/44790691/e0fa7404-6c4d-4e6f-8dad-fe707f53efef">

To summarize, we have the following options:
1. get rid of the arrow down, and bring back `/` and magnifying glass (or change style to match the more widely known "Type / to search" style). 
2. place a permanent hint in the main bar (left of connect button)
3. place a permanent hint in the main bar (in the center)
4. don't show any hint, treat the `/` as a hidden feature.

I personally support the last option. It reminds me a lot of github's "dot" shortcut (on any repository, you can press `.` to open an interactive vscode environment and actually run the project!). It's not advertised anywhere, but I heard it from a tiktok video, and use it every since. Truly a hidden feature thats badass, and knowing its hidden makes it even cooler. To me the shortcut search is exactly the same, an awesome hidden feature I don't want to miss, but no need to promote it prominently.

dcts commented:
Option 1 could be something like this (designer needed for proper mockup):
<img width="552" alt="Bildschirmfoto 2024-02-20 um 19 59 07" src="https://github.com/DeXter-on-Radix/website/assets/44790691/7fbe393b-8cc6-4849-b2f1-3bc39ef7c4f8">

dcts commented:
@nguvictor yes that makes sense! Adapted the PR so now `DEXTR/XRD` is the first pair.

dcts commented:
@EvgeniiaVak added a permanent hint as proposed in Option 1.

dcts commented:
@EvgeniiaVak good call! Fixed it, now the `/` is also working inside the search input. 

What do you think about the permanent shortcut hint that is currently placed on top right of the screen? Happy to change if you have a strong preference towards another option. If we have no clear solution we could also consult with the community via poll on telegram.
![permanent-hint](https://github.com/DeXter-on-Radix/website/assets/44790691/821117d2-4089-4e2e-bece-88bebd0977ff)

EvgeniiaVak commented:
@dcts my strong opinion, but very loosely held 😄 I'd prefer:
1. the way it is on mainnet now - the shortcut shown directly in the input field
2. or hidden if the first option for some reason is bad

if we keep the hint where it is, it probably should be less rounded?

about text colors - I noticed even the token1 color is darker now, is it from a design file? (looks worse in my opinion) 
![Screenshot 2024-02-22 at 15 13 16](https://github.com/DeXter-on-Radix/website/assets/27793901/abba24a2-5707-49ab-b3dd-f26216b4fc7e)

background for the down arrow looks a bit off

we probably need `cursor-pointer` on hovering this element because it's clickable

Do we have a related design file somewhere? Can we pin it in description here?

dcts commented:
@EvgeniiaVak ok sounds good. I'll ask in Telegram for a designer to take a look into this and iterate over current design (to make it more clear). For that goal, would you mind if we change the shortcut to for example Ctrl+K (CMD+K)? I think having 2 keys shown will make it very clear (see my previous suggestion, but ultimately up to the designer):
> Option 1 could be something like this (designer needed for proper mockup): <img alt="Bildschirmfoto 2024-02-20 um 19 59 07" width="552" src="https://private-user-images.githubusercontent.com/44790691/306385210-7fbe393b-8cc6-4849-b2f1-3bc39ef7c4f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg2MTg1ODEsIm5iZiI6MTcwODYxODI4MSwicGF0aCI6Ii80NDc5MDY5MS8zMDYzODUyMTAtN2ZiZTM5M2ItOGNjNi00ODQ5LWIyZjEtM2JjMzllZjdjNGY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIyVDE2MTEyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4N2NiNTVlYTdkMDg1ZmY5Y2ViOTk2ODRiNzY0MGZiNjgzYTkzZjRjMGEyYWY5N2U0MzMxMzM2MjhhMzdhNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vCG7YLnPKr1hLGbuN5sloIzNU1VsXX6EUxLQMnyoxaI">

The design file I used is this one (but @Kafkafrate adapted the designs recently, so you might not find the old designs, but I am actively working on the update):
https://www.figma.com/file/qushS15h87wjpoXaY6ND4W/DeXter%20UI%20-%20refresh?type=design&fuid=708572579352018658

dcts commented:
Adapted to @Kafkafrate's new design:
- 2 icons showing both coins
- added price on the right side. Did not add 24h price change as all coins besides DEXTER have invalid values (-1). I suspect this means no price change in last 24h for most coins. I suggest leaving only the price there for now and implement future improvements in a seperate CL, as we are not aligned on it yet.

@EvgeniiaVak I reverted the chevron-down to what we currently have.
